.github/workflows/main.yaml

```diff
@@ -19,14 +19,18 @@ jobs:
         java-version: 11
         distribution: 'adopt'
         cache: 'maven'
+        gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
+        gpg-passphrase: MAVEN_GPG_PASSPHRASE
```

Comment:
Why don't u use `${{ secrets.MAVEN_GPG_PASSPHRASE }}`directly here?

Reply:
Main reason is to follow the official example https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#publishing-using-apache-maven
If I understand well, it allows using the secret in a single step and avoiding to write it in the settings.xml file.